tests/login.test.js

```diff
@@ -0,0 +1,57 @@
+import LoginPage from '../pages/login.page';
+
+describe('Login Test Cases', () => {
+
+    it('Should display login page when browsing to the route', async () => {
+      await LoginPage.open('login');
+      expect(browser).toHaveUrlContaining('login');
+    });
+
+    it('Should log into demo account when valid data is entered', async () => {
+     
+      await LoginPage.open('login');
+      await LoginPage.completeFields(LoginPage.usernameInput, 'jsmith@demo.io');
+      await LoginPage.completeFields(LoginPage.passwordInput, 'Demo123!');
+      await LoginPage.logInWithSubmitButton();
+      expect(browser).toHaveUrlContaining('home');
+
+    });
+
+    it('Should log into demo account when using enter key', async () => {
+      await LoginPage.open('login');
+      await LoginPage.completeFields(LoginPage.usernameInput, 'jsmith@demo.io');
+      await LoginPage.completeFields(LoginPage.passwordInput, 'Demo123!');
+      await LoginPage.logInWithEnter();
+      expect(browser).toHaveUrlContaining('home');
+
+    });
+
+    it('Should deny access when invalid data is entered', async () => {
+      await LoginPage.open('login');
```

Comment:
I think you can encapsulate this a bit more, as it makes sense that each page knows the URL they are located at, so the test doesn't need to. Consider implementing a method that navigates to the login page (potentially using your current `open()` implementation under the hood). Something like `navigateToPage()`, so when you use it from the test level, the call looks like: `LoginPage.navigateToPage()` (without needing to pass a specific parameter).

`navigateToPage()` method would be implemented in `LoginPage` class like:

```javascript

async navigateToPage(){
    await this.open('login')
}
````